dango/dex/src/execute.rs

```diff
@@ -84,6 +80,165 @@ fn batch_update_pairs(ctx: MutableCtx, updates: Vec<PairUpdate>) -> anyhow::Resu
     Ok(Response::new().add_subevents(events))
 }
 
+#[inline]
+fn batch_update_orders(
+    mut ctx: MutableCtx,
+    creates: Vec<CreateLimitOrderRequest>,
+    cancels: Option<OrderIds>,
+) -> anyhow::Result<Response> {
+    let mut deposits = Coins::new();
+    let mut refunds = Coins::new();
+    let mut events = Vec::new();
+
+    // --------------------------- 1. Cancel orders ----------------------------
+
+    // First, collect all orders to be cancelled into memory.
+    let orders = match cancels {
+        // Cancel all orders.
+        Some(OrderIds::All) => ORDERS
+            .idx
+            .user
+            .prefix(ctx.sender)
+            .range(ctx.storage, None, None, IterationOrder::Ascending)
```

Comment:
could we ever conceivably run out of gas here? i.e. do we need to have pagination ability? would that be a problem in liquidation?

Reply:
actually there's gas consumption when reading data from DB, but computations in the VM don't have gas consumption yet.